tests/unit/ssl.c

```diff
@@ -0,0 +1,112 @@
+/*  vim:expandtab:shiftwidth=2:tabstop=2:smarttab:
+ *
+ *  Drizzle Client & Protocol Library
+ *
+ * Copyright (C) 2018 Sociomantic Tsunami
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ *     * The names of its contributors may not be used to endorse or
+ * promote products derived from this software without specific prior
+ * written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+#include <yatl/lite.h>
+#include <libdrizzle-redux/libdrizzle.h>
+
+/**
+ *  Test encrypted connections to a MySQL server
+ *
+ *  The key, certificate and certificate authority must have the default MySQL
+ *  SSL filenames
+ *   - client-key.pem
+ *   - client-cert.pem
+ *   - ca.pem
+ *
+ *  The path to the certificate and key files must be specified with the
+ *  environment variable DRIZZLE_MYSQL_CA_PATH
+ *
+ *  Actually establishing the encrypted connection the MySQL server is not
+ *  tested.
```

Comment:
This comment seems unclear to me when the connection is being created. Could you expand it a bit?

Reply:
Yes. It is basically that connecting using SSL to the MySQL server doesn't work without a workaround which disables non-blocking socket regardless of the options set by the client. Thus breaking the library for clients which uses non-blocking. Will rephrase it to make it more clear though